### ISSUE TITLE
Emit schema definition file

### ISSUE THREAD

**Issue by 19majkel94** [19majkel94/type-graphql#127]
## Creating SDL file from TypeGraphQL generated schema

Sometimes we need the schema to be printed into `schema.gql` file.
It's mainly needed for tooling like client-side queries autocompletion and validation.
It may also be used as a snapshot for detecting schema regression.
Other developer can also read the SDL file to explore the API instead of reading the complicated TypeGraphQL-based app code.

### Auto emit schema on build:
```ts
await buildSchema({
  resolvers: [__dirname + "./src/**/*.resolver.ts"],
  emitSchemaFile: true, // default to "schema.gql" in current work dir
  emitSchemaFile: "/path/to/folder/schema.gql", // or explicit path
});
```

### Programmatic usage:
```ts
import { emitSchemaDefinitionFile, buildSchema } from "type-graphql";
// ...

hypotheticFileWatcher.watch("./src/**/*.resolver.ts", async () => {
  const schema = getSchemaNotFromBuild();
  await emitSchemaDefinitionFile("/path/to/folder/schema.gql", schema);
});
```

**Comment by hughns** [comment]
In case it is helpful to anyone: we are using `get-schema` from `graphql-cli` to fulfill this requirement at the moment.

**Comment by 19majkel94** [comment]
@hughns It forces to launch a server (to `Download schema from endpoint`), you can just use `printSchema` from `graphql-js` which this feature is all about - not high end technology, just useful common tool with 3 lines of code 😄

**Comment by orta** [comment]
For folks who are interested, this is [how we do](https://github.com/artsy/kaws/blob/master/src/scripts/dumpSchema.ts) that (and apply prettier for consistency across projects):

```ts
import { writeFileSync } from "fs"
import { printSchema } from "graphql/utilities"
import { format } from "prettier"
import { createSchema } from "../createSchema"

const schemaPath = "_schema.graphql"

createSchema().then(schema => {
  // commentDescriptions means it uses # instead of the ugly """
  const schemaText = printSchema(schema, { commentDescriptions: true })
  const prettySchema = format(schemaText, { parser: "graphql" })

  // Save user readable type system shorthand of schema
  writeFileSync(schemaPath, prettySchema, "utf8")
})

```

**Comment by 19majkel94** [comment]
Released in `0.15.0`! 🚀
